aiida/backends/tests/cmdline/commands/test_computer.py

```diff
@@ -1,13 +1,357 @@
+    valid_noninteractive_options['hostname'] = "localhost"
+    valid_noninteractive_options['description'] = "my description"
+    ## Specifically give a string value for interactive prompts
+    if not non_interactive:
+        valid_noninteractive_options['enabled'] = "True"
+    valid_noninteractive_options['transport'] = "local"
+    valid_noninteractive_options['scheduler'] = "direct"
+    valid_noninteractive_options['shebang'] = "#!/bin/bash"
+    valid_noninteractive_options['work-dir'] = "/scratch/{username}/aiida_run"
+    valid_noninteractive_options['mpirun-command'] = "mpirun -np {tot_num_mpiprocs}"
+    valid_noninteractive_options['mpiprocs-per-machine'] = "2"
+    # Make them multiline to test also multiline options
+    valid_noninteractive_options['prepend-text'] = "date\necho 'second line'"
+    valid_noninteractive_options['append-text'] =  "env\necho '444'\necho 'third line'"
+
+    # I replace kwargs here, so that if they are known, they go at the right order
+    for k in replace_args:
+        valid_noninteractive_options[k] = replace_args[k]
+
+    return valid_noninteractive_options
+
+def generate_setup_options(ordereddict):
+    """
+    Given an (ordered) dict, returns a list of options
+
+    Note that at this moment the implementation only supports long options
+    (i.e. --option=value) and not short ones (-o value).
+    Set a value to None to avoid the '=value' part.
+
+    :param ordereddict: as generated by ``generate_setup_options_dict()``
+    :return: a list to be passed as command-line arguments.
+    """
+    options = []
+    for k, v in ordereddict.iteritems():
+        if v is None:
+            options.append('--{}'.format(k))
+        else:
+            options.append('--{}={}'.format(k, v))
+    return options
+
+def generate_setup_options_interactive(ordereddict):
+    """
+    Given an (ordered) dict, returns a list of options
+
+    Note that at this moment the implementation only supports long options
+    (i.e. --option=value) and not short ones (-o value).
+    Set a value to None to avoid the '=value' part.
+
+    :param ordereddict: as generated by ``generate_setup_options_dict()``
+    :return: a list to be passed as command-line arguments.
+    """
+    options = []
+    for k, v in ordereddict.iteritems():
+        if v is None:
+            options.append(True)
+        else:
+            options.append(v)
+    return options
+
+class TestVerdiComputerSetup(AiidaTestCase):
+    """Tests for the 'verdi computer setup' command."""
+
+    def setUp(self):
+        self.runner = CliRunner()
+
+    def test_help(self):
+        self.runner.invoke(setup_computer, ['--help'])
+
+    def test_reachable(self):
+        import subprocess as sp
+        output = sp.check_output(['verdi', 'computer', 'setup', '--help'])
+        self.assertIn('Usage:', output)
+
+    # TODO: test that mpiprocs-per-machine is not requested e.g. for SGE
```

Comment:
Make this into an issue instead.

Reply:
Actually it looks like you have a test for this already!

Reply:
Yes, I've a couple of leftover comments, I'm going to clean these up